Offroad-iOS/Offroad-iOS/Presentation/Diary/ViewController/MonthPickerModalViewController.swift

```diff
@@ -0,0 +1,178 @@
+//
+//  MonthPickerModelViewController.swift
+//  Offroad-iOS
+//
+//  Created by 조혜린 on 3/11/25.
+//
+
+import UIKit
+
+import RxSwift
+
+final class MonthPickerModalViewController: UIViewController {
+    
+    // MARK: - Properties
+    
+    private let rootView = MonthPickerModalView()
+    
+    private let (minYear, minMonth) = MyDiaryManager.shared.fetchYearMonthValue(dateType: .minimum)
+    private let (maxYear, maxMonth) = MyDiaryManager.shared.fetchYearMonthValue(dateType: .maximum)
+    private let (currentYear, currentMonth) = MyDiaryManager.shared.fetchYearMonthValue(dateType: .current)
+    
+    private lazy var selectedYear = currentYear
+    private lazy var selectedMonth = currentMonth
+
+    private lazy var years = Array(minYear...maxYear)
+    private lazy var months: [Int] = {
+        if currentYear == maxYear {
+            return Array(1...maxMonth)
+        } else if currentYear == minYear {
+            return Array(minMonth...12)
+        } else {
+            return Array(1...12)
+        }
+    }()
+    
+    // MARK: - Life Cycle
+    
+    override func loadView() {
+        view = rootView
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        setupDelegate()
+        setupTarget()
+        setupPickerView()
+    }
+    
+    override func viewWillLayoutSubviews() {
+        rootView.monthPickerView.subviews[1].isHidden = true
+    }
+}
+
+private extension MonthPickerModalViewController {
+    
+    // MARK: - Private Method
+    
+    func setupDelegate() {
+        rootView.monthPickerView.delegate = self
+        rootView.monthPickerView.dataSource = self
+    }
+    
+    func setupTarget() {
+        rootView.completeButton.addTarget(self, action: #selector(completeTapped), for: .touchUpInside)
+    }
+    
+    func setupPickerView() {
+        let selectedYearRow = years.firstIndex(of: currentYear)!
+        let selectedMonthRow = months.firstIndex(of: currentMonth)!
+        
+        selectedYear = years[selectedYearRow]
+        selectedMonth = months[selectedMonthRow]
+        
+        DispatchQueue.main.async {
+            self.rootView.monthPickerView.selectRow(selectedYearRow, inComponent: 0, animated: false)
+            self.rootView.monthPickerView.selectRow(selectedMonthRow, inComponent: 1, animated: false)
+        }
+    }
+}
+    
+@objc private extension MonthPickerModalViewController {
+
+    // MARK: - @objc Method
+    
+    func completeTapped() {
+        var dateComponents = DateComponents()
+        dateComponents.year = selectedYear
+        dateComponents.month = selectedMonth
+        
+        let selectedDate = Calendar.current.date(from: dateComponents)!
```

Comment:
P3
여기서 강제 언래핑을 사용한다면 nil이 반환될 때 앱이 강제로 종료될 수 있지 않을까요? 아래처럼 바꾸는 게 더 좋지 않을까 생각하는데요, 혹시 nil이 변환될 경우가 없어서 강제 언래핑을 사용하신 건 지 궁금합니다! 의도 설명해주시면 감사하겠습니다:)
```suggestion
        if let selectedDate = Calendar.current.date(from: dateComponents) {
        MyDiaryManager.shared.updateCalenderCurrentPage.accept(selectedDate)
    }
```

Reply:
완료 버튼을 누르는 시점에서는 항상 selectedYear, selectedMonth가 존재하기 때문에 dateComponents가 nil인 경우는 없다고 생각하여 강제 언래핑을 사용했으나, 혹시 모를 상황에 대비하여 if let 구문으로 변경하는 것도 좋아보이네요! 반영하겠습니다~